### PR TITLE
LTP: Allow to reboot after each test (LTP_REBOOT_AFTER_TEST=1)

### PR DESCRIPTION
**Pull request by pevik**
Add LTP_REBOOT_AFTER_TEST=1 variable to allow to reboot after each test (needed for certain tests, e.g. IMA).

Related ticket: https://progress.opensuse.org/issues/159045

Verification run: 
* https://openqa.suse.de/tests/overview?build=pevik%2Fos-autoinst-distri-opensuse%23LTP_REBOOT_AFTER_TEST 
   (NOTE: `ltp_ima_reboot` and `ltp_ima_selinux` fail on sle micro due missing support for installing LTP from git)
* https://openqa.opensuse.org/tests/overview?build=pevik%2Fos-autoinst-distri-opensuse%23LTP_REBOOT_AFTER_TEST

### CODE REVIEW

File: tests/kernel/install_ltp.pm [hunk 1]
```diff
@@ -521,6 +521,10 @@ will be used.
 
 Overrides the official LTP GitHub repository URL.
 
+=head2 LTP_INSTALL_REBOOT
+
```
Comment (czerw) [inline tests/kernel/install_ltp.pm:525]: This is just adding doc to existing feature, right?

Comment (pevik) [reply tests/kernel/install_ltp.pm:525]: Yes.

**Comment by pevik** [comment]
> Do we need `reboot` even for skipped tests https://openqa.suse.de/tests/16025349#step/ima_keys/12 ?

Good point, thanks! Solved by checking if return value != 32 (TCONF).
https://openqa.suse.de/tests/16048765#step/ima_keys/7

**Comment by pevik** [comment]
Rewritten detection of the last test (it did not work when LTP_COMMAND_EXCLUDE contained the latest item. I suspect it will not work when the latest item has skip in the yaml file, but I consider that minor).

**Review by czerw — APPROVED** [review]

File: tests/kernel/run_ltp.pm [hunk 2]
```diff
@@ -11,15 +11,30 @@ use 5.018;
...
+use LTP::utils 'prepare_ltp_env';
 use LTP::WhiteList;
 require bmwqemu;
 
+sub do_reboot {
+    my $self = shift;
+
+    record_info("reboot");
+    power_action('reboot', textmode => 1, keepconsole => is_pvm);
+    reconnect_mgmt_console if is_pvm;
```
Comment (mdoucha) [inline tests/kernel/run_ltp.pm:32]: `reconnect_mgmt_console` is also needed in baremetal tests: `get_var('LTP_BAREMETAL')`

**Review by mdoucha — CHANGES_REQUESTED** [review]

File: tests/kernel/run_ltp.pm (hunk 2 above)
Comment (pevik) [reply tests/kernel/run_ltp.pm]: Thanks fixed.

File: lib/LTP/utils.pm [hunk 3]
```diff
@@ -404,20 +404,25 @@ sub parse_openposix_runfile {
...
-        my $tinfo = testinfo($test_result_export, test => $test, runfile => $name);
 
+        my $test = {name => $1 . $suffix, command => $2, last => 0};
         if ($test->{name} =~ m/$cmd_pattern/ && !($test->{name} =~ m/$cmd_exclude/)) {
-            loadtest_runltp($test->{name}, $tinfo, $whitelist);
+            push @tests, $test;
         }
     }
+
+    ${tests [-1]}->{last} = 1;
```
Comment (mdoucha) [inline lib/LTP/utils.pm:421]: I guess we also need to check here whether `@tests` is empty, otherwise the test module may crash.

Comment (pevik) [reply lib/LTP/utils.pm]: Indeed:
```
Modification of non-creatable array value attempted, subscript -1 at
```
Fixed.

**Review by mdoucha — APPROVED** [review]

File: lib/LTP/utils.pm (hunk 3 above)
Comment (mdoucha) [reply lib/LTP/utils.pm]: `return unless @tests` would have been cleaner but this also works. Approved.
